### ISSUE TITLE
Enable SyncBatchnorm For Pytorch Lightning

### ISSUE THREAD

**Issue by lsy643** [Lightning-AI/lightning#13749]
## 🚀 Feature

Pytorch Lightning Lite is getting quite popular in our company, and many people hope sync batch norm can be supported when Lite is used.

It seems that we can just open it by passing the arg to the accelerator connector, so I create a PR [13748](https://github.com/Lightning-AI/lightning/pull/13748)  to enable it. 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

**Comment by tchaton** [comment]
Hey @[lsy643](https://github.com/lsy643), happy to hear LightningLite is useful to you ;)

We are investigating the idea of extracting accelerators and LightningLite on the same occasion from PyTorch Lightning to reduce dependencies and increase adoption.

What are your thoughts on the above?

Thanks for opening a PR ;)

**Comment by awaelchli** [comment]
This is a great feature. 
However, it is not as straightforward as passing down the bool flag to the accelerator connector. The way layer sync is applied in the strategies only works with the Trainer and LightningModule. 

In Lite, this probably has to be implemented on the level of self.setup() and the strategies have to be adapted/generalized for this.

**Comment by lsy643** [comment]
> This is a great feature. However, it is not as straightforward as passing down the bool flag to the accelerator connector. The way layer sync is applied in the strategies only works with the Trainer and LightningModule.
> 
> In Lite, this probably has to be implemented on the level of self.setup() and the strategies have to be adapted/generalized for this.

Thanks. The layer_sync is applied in the `setup` function in the ddp strategy, which seems not to be called in the `_setup_model_and_optimizers` function of the strategy. 

Is it appropriate to just add it directly in the `_setup_model` function?

And I am confused about the relationship about `setup`, '_setup_model' and 'configure_ddp'. Would you mind help me to clarify these things？

Thanks a lot

**Comment by awaelchli** [comment]
@lsy643 Thanks for starting to work on this and being patient with us. We can resume this work soon. We need to change the approach slightly from your initial PR, but it will be quite easy. For Lite to remain as flexible as it can be, we will want to introduce the `sync_batchnorm=True|False` as an argument to `Lite.setup()`. This would then wrap the layers for the model passed in. Furthermore, we should provide a teardown method to undo the sync-batchnorm layers #14682. These can be done in two separate steps.
